### PR TITLE
Prototype the `pipeline_executor_actor` interface

### PR DESCRIPTION
**Pull request by dominiklohmann**
Putting a logical pipeline into an actor interface requires making logical pipelines (and as such logical operators) copyable. This tests that to the point of simply making it possible to compile VAST with such an actor interface with tests yet to be written.

### CODE REVIEW

File: libvast/src/logical_pipeline.cpp [hunk 1]
```diff
@@ -281,6 +281,34 @@ auto make_local_producer(std::span<logical_operator_ptr> ops,
...
+
+logical_pipeline& logical_pipeline::operator=(const logical_pipeline& rhs) {
+  if (this != &rhs) {
+    ops_.clear();
+    ops_.reserve(rhs.ops_.size());
+    for (const auto& op : ops_)
+      ops_.push_back(op ? op->copy() : nullptr);
+  }
+  return *this;
+}
```
Comment (jachris) [inline libvast/src/logical_pipeline.cpp:301]: This violates our invariant that a `logical_pipeline` does not contain another `logical_pipeline`. Furthermore, we currently prohibit `nullptr` operators in `make`, and I would personally prefer to maintain this invariant everywhere. We can perhaps throw an exception if `copy()` returns a `nullptr`.

File: libvast/src/logical_operator.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+auto runtime_logical_operator::done() const noexcept -> bool {
+  return false;
+}
+
+auto runtime_logical_operator::copy() const noexcept -> logical_operator_ptr {
+  const auto repr = to_string();
+  auto result = logical_pipeline::parse(repr);
+  if (not result)
+    VAST_WARN("failed to copy logical pipeline '{}': {}", repr, result.error());
+  return std::make_unique<logical_pipeline>(std::move(*result));
```
Comment (jachris) [inline libvast/src/logical_operator.cpp:28]: This probably needs a `return nullptr` after `VAST_WARN`.

File: libvast/src/logical_operator.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+#include "vast/logical_operator.hpp"
+
+#include "vast/logical_pipeline.hpp"
+
+namespace vast {
+
+auto runtime_logical_operator::detached() const noexcept -> bool {
+  return false;
+}
```
Comment (jachris) [inline libvast/src/logical_operator.cpp:17]: This implementation does not match the documentation: `Returns whether any of the logical pipeline's operators are detached.`

File: libvast/src/logical_pipeline.cpp [hunk 4]
```diff
@@ -367,6 +395,13 @@ auto logical_pipeline::output_element_type() const noexcept
   return ops_.back()->output_element_type();
 }
 
+auto logical_pipeline::detached() const noexcept
+  -> bool {
+  return std::any_of(ops_.begin(), ops_.end(), [](const auto& op) noexcept {
+    return op ? op->detached() : false;
+  });
```
Comment (jachris) [inline libvast/src/logical_pipeline.cpp:402]: This can also be changed if we know that `op != nullptr`.

**Review by jachris — APPROVED** [review]

File: libvast/src/logical_pipeline.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/logical_pipeline.cpp:301]: Good catch, I'll simplify the logic and flatten here.

File: libvast/src/logical_operator.cpp (hunk 3 above)
Comment (jachris) [reply libvast/src/logical_operator.cpp:17]: Oh, and I even commented on it afterwards... Nevermind.

File: libvast/src/logical_operator.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/src/logical_operator.cpp:28]: I made this return an expected so we can choose to print the error at the call site instead of warning.

**Comment by dominiklohmann** [comment]
We discussed a different approach to doing `inspect` overloads internally. I will open a follow-up PR that takes care of this sometime this week, ideally.

**Pull request closed without merging**